src/mutations/authentication.ts

```diff
@@ -0,0 +1,81 @@
+import { QueryClient } from 'react-query';
+import Cookies from 'js-cookie';
+import { SUCCESS_MESSAGES } from '@graasp/translations';
+import * as Api from '../api';
+import {
+  signOutRoutine,
+  signInRoutine,
+  signInWithPasswordRoutine,
+  signUpRoutine,
+} from '../routines';
+import { CURRENT_MEMBER_KEY, MUTATION_KEYS } from '../config/keys';
+import { QueryClientConfig } from '../types';
+import { COOKIE_SESSION_NAME } from '../config/constants';
+
+export default (queryClient: QueryClient, queryConfig: QueryClientConfig) => {
+  const { notifier } = queryConfig;
+
+  queryClient.setMutationDefaults(MUTATION_KEYS.SIGN_IN, {
+    mutationFn: (payload) => Api.signIn(payload, queryConfig),
+    onSuccess: () => {
+      notifier?.({
+        type: signInRoutine.SUCCESS,
+        payload: { message: SUCCESS_MESSAGES.SIGN_IN },
+      });
+    },
+    onError: (error) => {
+      notifier?.({ type: signInRoutine.FAILURE, payload: { error } });
+    },
+  });
+
+  queryClient.setMutationDefaults(MUTATION_KEYS.SIGN_IN_WITH_PASSWORD, {
+    mutationFn: (payload) => Api.signInWithPassword(payload, queryConfig),
+    onSuccess: () => {
+      notifier?.({
+        type: signInWithPasswordRoutine.SUCCESS,
+        payload: { message: SUCCESS_MESSAGES.SIGN_IN_WITH_PASSWORD },
+      });
+      queryClient.resetQueries();
```

Comment:
Why is `resetQueries` only called for "sign in with password" and "sign out" but not for "sign in" and "sign up"?